packages/manager/apps/pci-volume-backup/src/data/hooks/useVolumeBackups.tsx

```diff
@@ -148,7 +148,7 @@ export function useVolumeBackups<T extends Record<string, unknown>>({
     flattenData: flattenData?.slice(0, (pageIndex + 1) * pageSize),
     isError,
     isLoading,
-    hasNextPage: pageIndex * pageSize + pageSize <= flattenData?.length,
+    hasNextPage: pageIndex * pageSize + pageSize < flattenData?.length,
```

Comment:
You should use a `useInfiniteQuery` hook instead that automatically returns `hasNextPage`

Reply:
That will remove all boilerplate code from your hook :)